projects/vdk-plugins/vdk-data-sources/src/vdk/plugin/data_sources/auto_generated.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2021-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from datetime import datetime
+from typing import Iterator
+from typing import List
+
+from vdk.plugin.data_sources.config import config_class
+from vdk.plugin.data_sources.config import config_field
+from vdk.plugin.data_sources.data_source import DataSourcePayload
+from vdk.plugin.data_sources.data_source import IDataSource
+from vdk.plugin.data_sources.data_source import (
+    IDataSourceConfiguration,
+)
+from vdk.plugin.data_sources.data_source import IDataSourceStream
+from vdk.plugin.data_sources.factory import data_source
+from vdk.plugin.data_sources.state import IDataSourceState
+
+
+@config_class(
+    name="auto_generated", description="Configuration for Auto generated data source"
+)
+class AutoGeneratedDataSourceConfiguration(IDataSourceConfiguration):
```

Comment:
Why did you not use just one class that ppl have to implement instead of 3? 
Would ppl reuse configurations with different streams of vis versa? 
Could you give an example of this?

Reply:
Nothing stops people from doing 

```
MyDataSource(IDataSourceConfiguration, IDataSource, IDataSourceStream): 
  ....
```

They really ned to register one class in both cases. 


It seems personal preference. It's easier for me to think them when they are separate as separate steps that need to be taken. Define configuration. Split your data into streams (at least 1) , write the stream reading logic